performance_test/include/performance_test/performance_node_base.hpp

```diff
@@ -219,6 +252,7 @@ class PerformanceNodeBase
   // Client blocking call does not work with timers
   // Use a lock variable to avoid calling when you are already waiting
   std::atomic<bool> m_client_lock {false};
+  std::atomic<bool> m_action_client_lock {false};
```

Comment:
if i recall correctly, this assumes that you are only doing 1 request from each node.
i think that each client should have their own "lock", rather than 1 per node